Server/src/main/java/org/xdi/oxauth/authorize/ws/rs/AuthorizeAction.java

```diff
@@ -332,9 +332,19 @@ public void checkPermissionGranted() {
     }
 
     private SessionId handleAcrChange(SessionId session, List<Prompt> prompts) {
-        if (session != null && prompts.contains(Prompt.LOGIN)) { // change session state only if prompt=none
+        if (session != null) {
             if (session.getState() == SessionIdState.AUTHENTICATED) {
-                session.getSessionAttributes().put("prompt", prompt);
+
+                String sessionPrompts;
+                if (prompts.contains(Prompt.LOGIN)) {
+                    sessionPrompts = prompt;
+                } else {
+                    prompts.add(Prompt.LOGIN);
+                    sessionPrompts = session.getSessionAttributes().get("prompt");
+                    sessionPrompts = StringUtils.isEmpty(sessionPrompts) ? "" : (sessionPrompts + " ");
+                    sessionPrompts = sessionPrompts + Prompt.LOGIN.getParamName();
+                }
+                session.getSessionAttributes().put("prompt", sessionPrompts);
```

Comment:
It seems it do the work but maybe we can replace these 11 lines with these 4 ?
```
if (!prompts.contains(Prompt.LOGIN)) {
    prompts.add(Prompt.LOGIN);
}
session.getSessionAttributes().put("prompt", StringUtils.implode(prompts, " ");
```

Reply:
The analog question of above applies here @yuriyz ...